mundane-dsv/src/test/scala/com/ambiata/mundane/dsv/DelimitedValuesParserSpec.scala

```diff
@@ -0,0 +1,73 @@
+package com.ambiata.mundane.dsv
+
+import com.ambiata.disorder._
+import Arbitraries._
+import org.scalacheck._, Gen._, Arbitrary._
+import org.specs2._
+import org.specs2.matcher.DisjunctionMatchers
+
+class DelimitedValuesParserSpec extends Specification with ScalaCheck with DisjunctionMatchers { def is = s2"""
+
+ A parser for delimited fields can deal with
+   comma-separated values $csv
+   quoted fields          $quoted
+   malformed lines        $malformed
+
+                                                                          """
+
+  def csv = prop { csvLine: CsvLine =>
+    DelimitedValuesParser.csv.parse(csvLine.value) must be_\/-(csvLine.expected)
+  }
+
+  def quoted = prop { csvLine: CsvQuotedLine =>
+    DelimitedValuesParser.csv.parse(csvLine.value) must be_\/-(csvLine.expected)
+  }
+
+  def malformed = prop { csvLine: CsvMalformedLine =>
+    DelimitedValuesParser.csv.parse(csvLine.value) must be_-\/
+  }
+
+  case class CsvLine(value: String, expected: List[String])
+
+  implicit def ArbitraryCsvLine: Arbitrary[CsvLine] =
+    Arbitrary(Gen.sized(n => listOfN(n, genRegularField).map(fs => CsvLine(fs.mkString(","), fs))))
+
+  case class CsvQuotedLine(value: String, expected: List[String])
```

Comment:
FWIW I wouldn't both fields here, just because one of them is derived and prints out if the test fails and can get in the way.